compiler/noirc_frontend/src/tests.rs

```diff
@@ -1214,6 +1214,17 @@ fn lambda$f1(mut env$l1: (Field)) -> Field {
         assert_eq!(get_program_errors(src).len(), 0);
     }
 
+    #[test]
+    fn operators_in_global_used_in_type() {
+        let src = r#"
+            global COUNT = 1 + 2;
```

Comment:
does this also work for nested globals, e.g.
```rust
global ONE = 1;
global TWO = 2;
global COUNT = 1 + 2;
```
?

Reply:
Did you mean to use `ONE` and `TWO` in `COUNT`?

Reply:
No, good catch. I've pushed a fix.

Now we need a fuel argument as well to avoid recursing forever in the case of cyclic globals.